### PR TITLE
UniStore: Fix DualWriter compare method

### PR DESCRIPTION
**Pull request by filewalkwithme**
**What is this feature?**

This PR fix the Dual Writer `Compare` function, which is currently broken (returning `true` for all comparions). 

This PR modifies the Compare function, modifying it to rely on the `Spec` field.

The reason for that is that LegacyStorage only know about `Spec`, and knows nothing about other Metadata values. Therefore, the `Compare` function must rely entirely on the `Spec` field.

**Why do we need this feature?**

We need to fix that to avoid that the Data Syncer tries to sync the entire collection of objects in Unified Storage.

**Who is this feature for?**

@grafana/grafana-search-and-storage 

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/search-and-storage-team/issues/115

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: pkg/apiserver/rest/dualwriter.go
Comment (suntala) [inline pkg/apiserver/rest/dualwriter.go:278]: Should we add a test case to [TestCompare](https://github.com/grafana/grafana/blob/main/pkg/apiserver/rest/dualwriter_test.go#L66) to prevent any regressions?

**Review by suntala — COMMENTED** [review]
LGTM
Might be worth adding a test to prevent a similar issue in the future.

File: pkg/apiserver/rest/dualwriter.go
Comment (filewalkwithme) [inline pkg/apiserver/rest/dualwriter.go:289]: Also, I realize now that `"spec"` field might not exists... I am going to add a test case for this as well

File: pkg/apiserver/rest/dualwriter.go
Comment (filewalkwithme) [reply pkg/apiserver/rest/dualwriter.go:277]: added!

File: pkg/apiserver/rest/dualwriter_test.go
Comment (suntala) [inline pkg/apiserver/rest/dualwriter_test.go:91]: If anything except spec is different, Compare will return true right? It might be worth reflecting that in the name of the test and make those fields different in the two objects being compared. So change not just kind and apiversion but also status and object meta (anything else?).

Comment (suntala) [inline pkg/apiserver/rest/dualwriter_test.go:91]: If anything except spec is different, Compare will return true right? It might be worth reflecting that in the name of the test and make those fields different in the two objects being compared. So change not just kind and apiversion but also status and object meta (anything else?).

**Review by suntala — APPROVED** [review]
Wondering if the test case covers enough, but I'll leave that up to you. Don't want to block the PR.

**Comment by suntala** [comment]
Oops didn't see you had automerge enabled.

**Pull request merged**
